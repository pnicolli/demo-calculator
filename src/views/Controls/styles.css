.controls {
    margin: 15px -2px 0;
    display: flex;
    flex-flow: column wrap;
    height: 250px;
}
