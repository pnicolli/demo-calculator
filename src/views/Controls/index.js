import React from 'react';
import { func } from 'prop-types';
import './styles.css';
import Button from '../../views/Button';

const Controls = ({
  resetState,
  resetCurrentValue,
  invertSign,
}) => (
  <div className="controls">
    {['0','1','2','3','4','5','6','7','8','9'].map((num, index) => (
      <Button onClick={() => {}} text={num} key={index} />
    ))}
  </div>
);

Controls.propTypes = {
  resetState: func,
  resetCurrentValue: func,
  invertSign: func,
};

export default Controls;
