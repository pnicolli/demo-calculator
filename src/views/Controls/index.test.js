import React from 'react';
import { shallow } from 'enzyme';
import Controls from './index';

const baseProps = {
  controls: [{ text: '+', onClick: () => {} }],
};

it('matches previous snapshot', () => {
  const tree = shallow(<Controls {...baseProps} />);
  expect(tree).toMatchSnapshot();
});
