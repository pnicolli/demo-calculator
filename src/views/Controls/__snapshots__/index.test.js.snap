// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches previous snapshot 1`] = `
<div
  className="controls"
>
  <Button
    onClick={[Function]}
    text="+"
  />
</div>
`;
